### ISSUE TITLE
Multiple language support for action bar messages, biomes and the user interface

### ISSUE THREAD

**Issue by rusthero** [rusthero/BiomeCompass#1]
I'm not a fan of making all messages configurable like in most plugins. Instead, I think it would be better to have multiple translations available and make them selectable through config.yml, like this:

```YAML
# Biome Compass Configuration File
# Check wiki to learn more https://github.com/rusthero/BiomeCompass/wiki

cooldown: 5
resolution: 32
radius: 6400
cacheSize: 100
...
language: en
```

This would provide server owners with a more streamlined and faster experience. I'm open to other suggestions and this is not a final proposal.

**Comment by Atharshoyeb** [comment]
Can u explain more about this issu i want to contibute

**Comment by rusthero** [comment]
Language files will be JSON files that are located in the `resources/lang` directory. These files will contain fields like the following:
```JSON
{
    "item.name": "Biomes Compass",

    "dimension.overworld": "Overworld",
    "dimension.nether": "Nether",
    "dimension.the_end": "Overworld",

    "menu.locate.title": "Select a biome",
    
    "actionbar.locating": "Locating",
    "actionbar.located": "Located",    

    "biome.ocean.name": "Ocean",
    "biome.plains.name": "Plains",
    "biome.taiga.name": "Taiga",
    "biome.old_growth_pine_taiga.name": "Old Growth Pine Taiga",
}
```
To access the selected language, we can use the `config.yml` and parse the language file for the selected language. For example, if the selected language is `en_us`, the language file would be located at `resources/lang/en_us.json`. 

All language fields will be the same for all languages, so using enums to store messages would be a good idea. Enums are a great way to store a fixed set of values, such as the messages in this case. This will make it easier to access and use the messages in the code.

**Comment by Atharshoyeb** [comment]
so i have to make yaml file for these

**Comment by rusthero** [comment]
Before creating JSON files for the language fields, it's important to first implement the ability to read from and parse those language files. This will ensure that your code can properly access and use the messages in the JSON files.

To do this, you can create a function that takes the selected language as an input and reads the corresponding JSON file from the resources/lang directory. The function can then parse the JSON file and store the messages in an enum, using the format shown in the original message.

For example:
```Java
import java.io.FileReader;
import java.io.IOException;
import java.util.Map;

import com.google.gson.Gson;

public class LanguageParser {
    public static Map<String, String> readLanguageFile(String language) throws IOException {
        String filePath = String.format("resources/lang/%s.json", language);
        FileReader reader = new FileReader(filePath);
        Map<String, String> messages = new Gson().fromJson(reader, Map.class);
        reader.close();
        return messages;
    }
}
```

To use this class, you can first create the JSON files for the different languages in the resources/lang directory. Then, you can call the readLanguageFile method, passing in the selected language as a parameter. The method will return a map containing the messages in the JSON file, which you can then use in your code.

Here is an example of how you might use this class:
```Java
String selectedLanguage = "en_us";
Map<String, String> messages = LanguageParser.readLanguageFile(selectedLanguage);
String itemName = messages.get("item.name");
String biomeOceanName = messages.get("biome.ocean.name");
```
I hope this helps! Let me know if you have any other questions.

**Comment by rusthero** [comment]
This feature may be a little hard to implement. I will work on this.

**Comment by rusthero** [comment]
I started working on this issue, language file for v1.3.0  will look like this for fellow translators: `lang/en_us.json`
```JSON
{
  "lang.name": "American English",
  "lang.name.transliteration": "English (US)",
  "lang.locale_code": "en_us",

  "log.biome_compass.enabled": "BiomeCompass is enabled",
  "log.biome_compass.disabled": "BiomeCompass is disabled",
  "log.vault.missing": "Money cost is disabled because Vault plugin is missing",
  "log.version_tracker.outdated": "I am outdated! The latest version is %s. Please update to ensure compatibility and access to new features",
  "log.version_tracker.error": "Version tracker could not check for the latest version",
  "log.player.locate": "%s requested to locate biome %s at location %d, %d, %d in %s",

  "item.biome_compass.display_name": "Biome Compass",
  "item.biome_compass.locating": "Locating",
  "item.biome_compass.not_found": "Not found within search range",
  "item.biome_compass.not_holding": "Hold the compass",
  "item.biome_compass.located": "Located",
  "item.biome_compass.located.distance": "%d blocks away",
  "item.biome_compass.on_cooldown": "Cooling down",
  "item.biome_compass.no_experience": "Insufficient experience",
  "item.biome_compass.no_funds": "Insufficient funds",
  "item.biome_compass.cost": "Cost",
  "item.biome_compass.cost.xp": "XP",
  "item.biome_compass.unknown_biome": "Unknown biome",

  "gui.biomes_menu.title.overworld": "OVERWORLD",
  "gui.biomes_menu.title.nether": "NETHER",
  "gui.biomes_menu.title.the_end": "THE END",
  "gui.biomes_menu.title.select": "Select a biome",
  "gui.biomes_menu.disallowed": "Not allowed in this dimension",

  "biome.ocean.name": "Ocean",
  "biome.plains.name": "Plains",
  "biome.desert.name": "Desert",
  "biome.windswept_hills.name": "Windswept Hills",
  "biome.forest.name": "Forest",
  "biome.taiga.name": "Taiga",
  "biome.swamp.name": "Swamp",
  "biome.mangrove_swamp.name": "Mangrove Swamp",
  "biome.river.name": "River",
  "biome.frozen_ocean.name": "Frozen Ocean",
  "biome.frozen_river.name": "Frozen River",
  "biome.snowy_plains.name": "Snowy Plains",
  "biome.mushroom_fields.name": "Mushroom Fields",
  "biome.beach.name": "Beach",
  "biome.jungle.name": "Jungle",
  "biome.sparse_jungle.name": "Sparse Jungle",
  "biome.deep_ocean.name": "Deep Ocean",
  "biome.stony_shore.name": "Stony Shore",
  "biome.snowy_beach.name": "Snowy Beach",
  "biome.birch_forest.name": "Birch Forest",
  "biome.dark_forest.name": "Dark Forest",
  "biome.snowy_taiga.name": "Snowy Taiga",
  "biome.old_growth_pine_taiga.name": "Old Growth Pine Taiga",
  "biome.windswept_forest.name": "Windswept Forest",
  "biome.savanna.name": "Savanna",
  "biome.savanna_plateau.name": "Savanna Plateau",
  "biome.badlands.name": "Badlands",
  "biome.wooded_badlands.name": "Wooded Badlands",
  "biome.warm_ocean.name": "Warm Ocean",
  "biome.lukewarm_ocean.name": "Lukewarm Ocean",
  "biome.cold_ocean.name": "Cold Ocean",
  "biome.deep_lukewarm_ocean.name": "Deep Lukewarm Ocean",
  "biome.deep_cold_ocean.name": "Deep Cold Ocean",
  "biome.deep_frozen_ocean.name": "Deep Frozen Ocean",
  "biome.sunflower_plains.name": "Sunflower Plains",
  "biome.windswept_gravelly_hills.name": "Windswept Gravelly Hills",
  "biome.flower_forest.name": "Flower Forest",
  "biome.ice_spikes.name": "Ice Spikes",
  "biome.old_growth_birch_forest.name": "Old Growth Birch Forest",
  "biome.old_growth_spruce_taiga.name": "Old Growth Spruce Taiga",
  "biome.windswept_savanna.name": "Windswept Savanna",
  "biome.eroded_badlands.name": "Eroded Badlands",
  "biome.bamboo_jungle.name": "Bamboo Jungle",
  "biome.meadow.name": "Meadow",
  "biome.grove.name": "Grove",
  "biome.snowy_slopes.name": "Snowy Slopes",
  "biome.frozen_peaks.name": "Frozen Peaks",
  "biome.jagged_peaks.name": "Jagged Peaks",
  "biome.stony_peaks.name": "Stony Peaks",
  "biome.dripstone_caves.name": "Dripstone Caves",
  "biome.lush_caves.name": "Lush Caves",
  "biome.deep_dark.name": "Deep Dark",
  "biome.nether_wastes.name": "Nether Wastes",
  "biome.soul_sand_valley.name": "Soul Sand Valley",
  "biome.crimson_forest.name": "Crimson Forest",
  "biome.warped_forest.name": "Warped Forest",
  "biome.basalt_deltas.name": "Basalt Deltas",
  "biome.the_end.name": "The End",
  "biome.small_end_islands.name": "Small End Islands",
  "biome.end_midlands.name": "End Midlands",
  "biome.end_highlands.name": "End Highlands",
  "biome.end_barrens.name": "End Barrens",
  "biome.the_void.name": "The Void"
}
```

**Comment by rusthero** [comment]
Feature is ready and going to be included in next release (v1.3.0).
